src/Extend/Theme.php

```diff
@@ -52,8 +54,61 @@ public function overrideFileSource(string $file, string $newFilePath, string $ex
         return $this;
     }
 
+    /**
+     * This method allows you to add custom Less functions.
+     *
+     * All custom Less functions may only return numbers, strings or booleans.
+     *
+     * **Example usage:**
+     * ```php
+     * (new Extend\Theme)
+     *     ->addCustomLessFunction('is-flarum', function (mixed $text) {
+     *         return strtolower($text) === 'flarum'
+     *     }),
+     * ```
+     *
+     * @see https://leafo.net/lessphp/docs/#custom_functions
+     *
+     * @param string $functionName Name of the function identifier.
+     * @param callable $callable The PHP function to run when the Less function is called.
+     * @return self
+     */
+    public function addCustomLessFunction(string $functionName, callable $callable): self
+    {
+        $this->customFunctions[$functionName] = function (...$args) use ($callable) {
+            $argVals = array_map(function ($arg) {
+                return $arg->value;
+            }, $args);
+
+            $return = $callable(...$argVals);
+
+            // Booleans
+            if (is_bool($return)) {
+                return new \Less_Tree_Quoted('', $return ? 'true' : 'false');
+            }
+
+            // Strings
+            if (is_string($return)) {
+                return new \Less_Tree_Quoted('', $return);
+            }
+
+            // Numbers
+            if (is_numeric($return)) {
+                return new \Less_Tree_Dimension($return);
+            }
+
+            throw new RuntimeException('Custom Less functions must only return a string, number or boolean.');
```

Comment:
Nitpick: maybe include the name of the function?

Reply:
```suggestion
            throw new RuntimeException('Custom Less function `'.$functionName.'` must only return a string, number or boolean.');
```